### ISSUE TITLE
[Feature Request] Sortable/Filter Options for Home View

### ISSUE THREAD

**Issue by rwardlow01** [futo-org/grayjay-android#78]
Could you please add the ability to add filters to the home view and/or make it sortable by newest videos first.

**Comment by dimyself** [comment]
Also for: downloads, playlists, and watch later please!

At minimum... Sorting (and also maybe searching) is needed for: downloads, playlists, and watch later.

These categories are almost unusable without proper sorting and a way to find videos. 

Downloads are only usable to me because I will download a video in my sub feed and click on it to watch from my sub feed... Forget about finding a downloaded video through the downloads tab

Same for playlists and watch later! I don't even use watch later because all the videos seem to be in a random order with no way to find anything. 

Playlist videos are the same... in a random order with no way to find videos other than manually scrolling / looking / hoping to find what you're looking for.

**Comment by Zvonimir-FUTO** [comment]
Just to add to this FR from #795, user requested search for **Downloads** and **Playlists**.

**Comment by dimyself** [comment]
I'm not sure if this request should go here or a different feature request  

Please also add multi select for Downloads (and playlists too?) so that it is easier to delete multiple videos at a time instead of deleting each video individually, which is time consuming.

**Comment by wedsa5** [comment]
I want to put in my suggestion for sorting. it currently looks like the Watch Later playlist is ordered by most recently added. I would like it to be sortable by least recently added. This is how YT sorts by default. 
When watching videos in Watch Later, you expect to watch them in the order that you added them to the playlist, not the reverse.

**Comment by SCBionicle** [comment]
This would be really handy to be able to change the watch later list from stack to FIFO ordering. It's really hard to watch my older bookmarks with how the watch later puts new videos in the front of the list rather than the end.

**Comment by dimyself** [comment]
> This would be really handy to be able to change the watch later list from stack to FIFO ordering. It's really hard to watch my older bookmarks with how the watch later puts new videos in the front of the list rather than the end.

Not just watch later...but downloads, and playlists.

It says no one is even assigned to this... 

It seems like this should have some kind of priority? 

It doesn't seem like it would be that hard to implement?

It makes the app hard to use without sorting in each category / folder / playlist

Also, another way to delete downloads other than manually deleting each download manually / individually is needed (like multi-select, and / or able to delete from the main feed, or long press menu with delete being one option)

**Comment by dairefagan** [comment]
Grayjay currently sorts Watch Later FIFO but on the YouTube website and app when manual sort is selected is LILO, so this should probably be the default, although I am posting to echo the request for the sort and filter options in the lists others have asked for - and for history.

Additionally, I have manually imported a 450+ Watch Later list by inserting the URLs into the store file before those I added using the app, so like the existing URLs, the new list remains in most recently added order ascending. They do populate Watch Later in the app but all of the newly imported items are sorted randomly every time I open the app, I assume because normally the timestamp they were added by the app is stored somewhere else and the app views these new items as unsortable.

As manually importing like this seems to be the only way to import to Watch Later without a YouTube account, can Grayjay please either respect the _reverse_ array order (LILO) as default or fall back on the order if necessary?

**Comment by dairefagan** [comment]
Grayjay currently sorts Watch Later FIFO but on the YouTube website and app when manual sort is selected LILO is used, so this should probably be the default, although I am posting to echo the requests for the sort and filter options in the lists others have asked for - and for History.

Additionally, as there is no way to import to Watch later by file or login (#1870), I have tired importing a playlist to this by appending the URLs to those that were already added to the stores file using the app, so like the existing URLs, the new array remains ordered by most recent. The imported videos do populate Watch Later in the app but unlike the app added items, which they are sometimes interleaved with, they are sorted randomly every time I open the app. I assume they are unsortable because normally when adding with the app a timestamp or some other ordering is recorded is a non export file.

As manually importing like this seems to be the only way to import to Watch Later right now, can Grayjay please either respect the order of the array as default (LILO), or use it as a fall back when unsortable items are added as described?

**Comment by dimyself** [comment]
Right? It seems like it should be pretty easy to implement

It makes using the app hard to use, cant sort your downloads, playlists,
watch later

**Comment by Zvonimir-FUTO** [comment]
Search and sorting added for Downloads in v284.

**Comment by wedsa5** [comment]
@Zvonimir-FUTO 

> Search and sorting added for Downloads in v284.

Thank you! are there plans to add it to watch later, ect?

**Comment by Zvonimir-FUTO** [comment]
yes yes, just didn't have enough time before release to pack all of those.

**Comment by dimyself** [comment]
> yes yes, just didn't have enough time before release to pack all of those.

thank you so much!!

**Comment by dimyself** [comment]
Would it be possible to have searching to also include based on channels
and not only based on video title? There are times when I know the channel
and not video title.

Currently if i search based on channel name, it doesn't work.

Thank you

**Comment by dairefagan** [comment]
> Would it be possible to have searching to also include based on channels
> and not only based on video title?

I too found it strange this was not the way it worked already and was going to request this also.

**Comment by Zvonimir-FUTO** [comment]
> Would it be possible to have searching to also include based on channels and not only based on video title?

Yes

**Comment by dimyself** [comment]
Thank you!!

**Comment by PhilC813** [comment]
Will this also get implemented for the playlists tab in channel pages?

I was trying to retrieve episodes of the "WAN Show" podcast from the [Linus Tech Tips channel](https://m.youtube.com/@LinusTechTips), and since the relevant playlist is old, you have to scroll super far to find it. The YouTube website makes this easier by having a dedicated "Podcasts" tab on the channel page, which isn't available on Grayjay.

**Comment by Zvonimir-FUTO** [comment]
> Will this also get implemented for the playlists tab in channel pages?

Yes we could change that view as well.

**Comment by NanNam226** [comment]
I would also call for sort and search features to be embedded in the add to playlist dropdown menu. I have way too many playlists to just scroll through them all every time I want to add content. 

Also, If I want to add a video to multiple playlists, I have to scroll through the list multiple times. A checkbox for adding and for visually showing that the video is already added should just be clickable without closing down the dropdown menu. 

In general, for search filtering and sorting, these are essential for every list-style view in the GrayJay app, especially since moving to GrayJay disengages the algorithm. This makes fine tuning the search and filtering parameters essential when trying to search for a specific video, be it in my own playlists, downloaded videos, watch later list, or just when searching for videos from the platforms themselves. And, of course, being able to order the lists of results, or just the entire list itself, is a core tool in any well-engineered search tool. F.x. to me, watch later is supposed to act like a stack, that orders the list according to when I added stuff, with the latest additions readily available on top of the stack.

I know that you are working on bringing algorithms to GrayJay, but even if they work really well, these tools are as important or maybe even more important, in my opinion.

**Comment by NanNam226** [comment]
Another thing I noticed. It would be good to standardize all filtering, sorting and search menus in how they behave. For Home, the filtering menu lets me do a long pres to single out one filter option, turning all other options off. This is not the case in the Subscriptions view/tab. 

These characteristics should just be defined once, somewhere, for consistency. 

To me the Home tab and the Subsccriptions tab should have equal opportunities for options in general, but I see how options or visual representation for a playlist might meaningfully be made to differ from that of Home and Subscriptions, like not having the 'watch later' button on videos displayed in the watch later list, or like visualizing the feature that lets you manually order a list. For the most part though, filtrering, sorting and searching options should function in the same way everywhere, just with different scopes of what is being filtered, sorted or searched in. 

Lists in playlist form should also be standardized everywhere in this way, as well as the drop down menus, where as of now, the add to part of the text for adding to playlists have been removed when opening the options for a video found under Subscriptions, while this is not the case in from the dropdown menu for videos found under playlists. 

Videos in watch later don't even have options for a dropdown menu, which also seems like it should be easily standardized as it is just another list like watch later or playlists. 

In general it seems that a lot of code is being written multiple times for these features, making changes inconsistent across the platform, and leading to a user experience of coming to expect something that then suddenly works differently in another part of the app.

Maybe standardization should be a seperate issue in itself, but it seems very relevant for the current issue of implementing filters, sorting and search features.
